### ISSUE TITLE
Add support for ObjectId (mongo)

### ISSUE THREAD

**Issue by jorisre** [blitz-js/superjson#69]
Hi 👋🏻  !

I'm using mongo with Next.js _(gsp & gssp)_ for a personal project and encounter the following error:
```sh
SerializableError: Error serializing `.json.recipes[0]._id` returned from `getStaticProps` in "/".
Reason: `object` ("[object Object]") cannot be serialized as JSON. Please only return JSON serializable data types.
    at isSerializable (/Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:9:7)
    at /Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:503
    at Array.every (<anonymous>)
    at isSerializable (/Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:304)
    at /Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:864
    at Array.every (<anonymous>)
    at isSerializable (/Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:768)
    at /Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:503
    at Array.every (<anonymous>)
    at isSerializable (/Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:304)
    at /Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:503
    at Array.every (<anonymous>)
    at isSerializable (/Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:7:304)
    at isSerializableProps (/Users/joris/w/peck/node_modules/next/dist/lib/is-serializable-props.js:9:219)
    at renderToHTML (/Users/joris/w/peck/node_modules/next/dist/next-server/server/render.js:28:992)
    at runMicrotasks (<anonymous>)
```

It look likes ObjectId isn't serializable: https://docs.mongodb.com/manual/reference/method/ObjectId/
In Next.js example, `_id` is converted to string : https://github.com/vercel/next.js/blob/canary/examples/with-mongodb-mongoose/pages/index.js#L58 but I don't want to do it each time.

```ts
const _id = new mongoose.Types.ObjectId();

// Object.getPrototypeOf(_id) returns ⬇

{
  toHexString: [Function (anonymous)],
  get_inc: [Function (anonymous)],
  getInc: [Function (anonymous)],
  generate: [Function (anonymous)],
  toString: [Function (anonymous)],
  toJSON: [Function (anonymous)],
  equals: [Function: equals],
  getTimestamp: [Function (anonymous)],
  generationTime: [Getter/Setter],
  [Symbol(nodejs.util.inspect.custom)]: [Function (anonymous)],
  [Symbol(mongoose#ObjectId)]: true
}
```

Do you think, SuperJson could handle that case ?

**Comment by Skn0tt** [comment]
Have you tried using `registerClass`? It's not documented, but can be used to serialise custom classes.
You'll need to make sure it's called both on the backend and in the frontend, maybe inside of your `_app.tsx`.

**Comment by jorisre** [comment]
I tried 
```ts
import { Types } from 'mongoose';
import superjson from 'superjson';

superjson.registerClass(Types.ObjectId);
```

But I still getting the same error.
I tried with and without `babel-plugin-superjson-next`

**Comment by Skn0tt** [comment]
I've put up a PR that adds support for this ☺️

**Comment by jorisre** [comment]
Thanks !

**Comment by Skn0tt** [comment]
Implemented in #71.

**Comment by italodeandra** [comment]
As a future reference:

If you don't want to import `mongodb` on the frontend (lower the bundle size) you should import the **ObjectId** from `bson`.

```ts
import { ObjectID } from "bson";
```

`bson` is a peer dependency from the `mongodb` driver and is where the **ObjectId** is from.
